snuba/utils/streams/streaming.py

```diff
@@ -374,6 +386,18 @@ def close(self) -> None:
         if self.__batch_builder is not None and len(self.__batch_builder) > 0:
             self.__submit_batch()
 
+    def terminate(self) -> None:
+        self.__closed = True
+
+        logger.debug("Terminating %r...", self.__pool)
+        self.__pool.terminate()
+
+        logger.debug("Shutting down %r...", self.__shared_memory_manager)
+        self.__shared_memory_manager.shutdown()
+
+        logger.debug("Terminating %r...", self.__next_step)
+        self.__next_step.terminate()
```

Comment:
Is this order intentional? Could __next_step try to access the shared memory between the `shutdown` of the memory manager and the completion of the call to `terminate` ? And would that be a problem ?